### PR TITLE
Add Retry Policy section to Jargon Mesh

### PR DESCRIPTION
**Pull request by djmagee**
## What does this PR do?

Adds the Retry Policy section, which is drawn from the [Retry Policy](https://docs.temporal.io/docs/concepts/workflows#retry-policy) section of the Workflows topic, the [Retries](https://docs.temporal.io/docs/concepts/activities#retries) section of the Activities topic, the [Retry Policy](https://www.notion.so/temporalio/Retry-Policy-06ddfe79b22a4b9392bbec9a935707b2) draft in Notion, and discussions with Dominik.

## Notes to reviewers

The definition of _retry interval_ merits close examination. We also discussed inserting a small diagram to illustrate the formula.

### CODE REVIEW

**Comment by djmagee** [comment]
I just looked at the [task](https://www.notion.so/temporalio/Add-Retry-Policy-to-the-terminology-page-e5dc7a696f3b4c96b5624883a582ff67) in Notion, which reminded me that I also need to link to this definition from other topics. I'll do that step Real Soon Now (i.e., mañana).

File: docs/jargon/mesh.md [hunk 1]
```diff
@@ -80,6 +80,69 @@ An [**Activity**](#activity) that is invoked directly in the same process by Wor
...
+## Retry Policy
+
+A collection of attributes that instructs Temporal Server how to retry a failure of an Activity Task Execution or a [**Workflow Execution**](#workflow-execution).
+Any custom Retry Policy must be provided when an [**Activity Task**](#activity-task) or Workflow starts.
+
+Retrying an [**Activity Task**](#activity-task) produces a new Activity Task Execution that uses the same parameters [VERIFY] and Task Queue as the previous attempt.
+
+An Activity Task Execution has a default Retry Policy and thus retries by default.
+
+A [**Workflow Execution**](#workflow-execution) does not have a default Retry Policy and thus does not retry by default.
```
Comment (flossypurse) [inline docs/jargon/mesh.md:92]:
```suggestion
[**Workflow Executions**](#workflow-execution) do not start with a default Retry Policy and thus does not retry by default.
```

File: docs/jargon/mesh.md [hunk 2]
```diff
@@ -80,6 +80,69 @@ An [**Activity**](#activity) that is invoked directly in the same process by Wor
...
+- A file-processing or media-encoding Workflow Execution that downloads files to a host.
+
+A Retry Policy does not apply to a Workflow Task Execution, which is governed by the Schedule-To-Start timeout.
+
+The time between retries is the _retry interval_. A retry interval is the smaller of two values:
+
+- The [initial interval](#initial-interval) multiplied by the [backoff coefficient](#backoff-coefficient) raised to power of the number of retries.
+- The [maximum interval](#maximum-interval).
+
+A Retry Policy contains the following settings:
```
Comment (flossypurse) [inline docs/jargon/mesh.md:108]:
```suggestion
A Retry Policy contains the following attributes:
```

File: docs/jargon/mesh.md [hunk 3]
```diff
@@ -80,6 +80,69 @@ An [**Activity**](#activity) that is invoked directly in the same process by Wor
...
+(Be aware that, by default, a Workflow Task Execution does retry indefinitely.)
+
+A [**Workflow Execution**](#workflow-execution) should be retried only in specific scenarios, such as the following:
+
+- A Cron Workflow or some other stateless always-running Workflow Execution that can benefit from retries.
+- A file-processing or media-encoding Workflow Execution that downloads files to a host.
+
+A Retry Policy does not apply to a Workflow Task Execution, which is governed by the Schedule-To-Start timeout.
+
+The time between retries is the _retry interval_. A retry interval is the smaller of two values:
```
Comment (flossypurse) [inline docs/jargon/mesh.md:103]:
```suggestion
The time between retries is the _retry interval_.
A retry interval is the smaller of two values:
```

Comment (flossypurse) [reply docs/jargon/mesh.md:103]: Shall we include the simple diagram here?

File: docs/jargon/mesh.md [hunk 4]
```diff
@@ -80,6 +80,69 @@ An [**Activity**](#activity) that is invoked directly in the same process by Wor
 
 - Although a Local Activity consumes less resources than a normal [**Activity**](#activity), it is subject to shorter durations and a lack of rate limiting.
 
+## Retry Policy
+
+A collection of attributes that instructs Temporal Server how to retry a failure of an Activity Task Execution or a [**Workflow Execution**](#workflow-execution).
+Any custom Retry Policy must be provided when an [**Activity Task**](#activity-task) or Workflow starts.
```
[the remark this answers is not in the document]
Comment (djmagee) [reply docs/jargon/mesh.md:86]: Good point. I'll ax the link for now. We'll need to retrofit links for a number of terms later anyway.

File: docs/jargon/mesh.md (hunk 3 above)
Comment (djmagee) [reply docs/jargon/mesh.md]: Makes sense to me. @dtornow?

**Comment by tsurdilo** [comment]
i think for max retries you are missing the case where you set it to 1 also if you set it to negative it raises error

**Comment by tsurdilo** [comment]
might also be missing Workflow.retry fron apis when twlking about retries. workflow options can also set retry options iirc not just activity options

**Comment by flossypurse** [comment]
> might also be missing Workflow.retry fron apis when talking about retries. workflow options can also set retry options iirc not just activity options

I am not familiar with a Retry API ?

File: docs/jargon/mesh.md [hunk 5]
```diff
@@ -80,6 +92,86 @@ An [**Activity**](#activity) that is invoked directly in the same process by Wor
 
 - Although a Local Activity consumes less resources than a normal [**Activity**](#activity), it is subject to shorter durations and a lack of rate limiting.
 
+## Retry Policy
+
+A collection of attributes that instructs the Temporal Server how to retry a failure of an [Activity Task Execution](#activity-task-execution) or a [Workflow Execution](#workflow-execution).
+
+- If a custom Retry Policy is to be used, it must be provided as an options parameter when an [Activity Execution](#activity-execution) or [Workflow Execution](#workflow-execution) is invoked.
+
+- The time between a retry is the _retry interval_.
```
Comment (djmagee) [inline docs/jargon/mesh.md:101]: I previously had "The time between retries...", but that leaves out the first retry interval, which is between the initial attempt and the first retry attempt. My next commit changes the phrase to "The wait time before a retry...."

File: docs/jargon/mesh.md [hunk 6]
```diff
@@ -80,6 +92,86 @@ An [**Activity**](#activity) that is invoked directly in the same process by Wor
...
+  A retry interval is the smaller of two values:
+  - The [Initial Interval](#initial-interval) multiplied by the [Backoff Coefficient](#backoff-coefficient) raised to the power of the number of retries.
+  - The [Maximum Interval](#maximum-interval).
+
+![retry interval](/img/retry-interval-diagram.png)
```
Comment (djmagee) [inline docs/jargon/mesh.md:106]: We need to add Maximum Interval to the formula in this diagram, @dtornow .

**Review by flossypurse — APPROVED** [review]

**Pull request merged**